addons/mail/static/tests/messaging_menu/messaging_menu.test.js

```diff
@@ -1226,6 +1226,12 @@ test("Can quick search when more than 20 items", async () => {
     await insertText(".o-mail-MessagingMenu input", "admin");
     await contains(".o-mail-NotificationItem", { count: 1 });
     await contains(".o-mail-NotificationItem", { text: "Mitchell Admin" });
+    await insertText(".o-mail-MessagingMenu input", "no threads", { replace: true });
+    await contains(".o-mail-MessagingMenu div.text-muted", { text: "No thread found." });
+    await contains(".o-mail-MessagingMenu div.text-muted", {
+        text: "No conversation yet...",
+        count: 0,
+    });
```

Comment:
Negative assertion without matching positive assertion is not good: test is too prone to pass when the behavior is still not ok. For example: rendering could happen with a delay (not likely) or the string could have changed (very likely, and this passes when this shouldn't).

I would suggest to await "No thread found." like in code (that's good), but then assert synchronously the whole content of messaging menu in the list part is "No thread found.".

This gives much more value that it shows this and only this on this screen... at least on immediate showing of this string! We cannot guarantee this is indefinitely that value and it's not worth making the test longer artificially for the sake of giving more guarantee, this isn't cheap for each test to be at least 3 seconds long!